api/src/modules/projects/projects.service.ts

```diff
@@ -21,14 +29,70 @@ export class ProjectsService extends AppBaseService<
 
   async extendFindAllQuery(
     query: SelectQueryBuilder<Project>,
-    fetchSpecification: FetchSpecification,
+    fetchSpecification: ExtendedFetchSpecification,
   ): Promise<SelectQueryBuilder<Project>> {
     // Filter by project name
     if (fetchSpecification?.filter?.projectName) {
       query = query.andWhere('project_name ILIKE :projectName', {
         projectName: `%${fetchSpecification.filter.projectName}%`,
       });
     }
+
+    // Filter by abatement potential
+    if (this.isAbatementPotentialFilterValid(fetchSpecification)) {
+      query = query.andWhere(
+        'abatement_potential >= :minAP AND abatement_potential <= :maxAP',
+        {
+          minAP: fetchSpecification.otherFilters.abatementPotentialRange[0],
+          maxAP: fetchSpecification.otherFilters.abatementPotentialRange[1],
+        },
+      );
+    }
+
+    // Filter by cost (total or NPV)
+    if (this.isCostFilterValid(fetchSpecification)) {
+      let filteredCostColumn: string;
+      switch (fetchSpecification.otherFilters.costRangeSelector) {
+        case 'npv':
+          filteredCostColumn = 'total_cost_npv';
+          break;
+        case 'total':
+        default:
+          filteredCostColumn = 'total_cost';
+          break;
+      }
+
+      query = query.andWhere(
+        `${filteredCostColumn} >= :minCost AND ${filteredCostColumn} <= :maxCost`,
+        {
+          minCost: fetchSpecification.otherFilters.costRange[0],
+          maxCost: fetchSpecification.otherFilters.costRange[1],
+        },
+      );
+    }
     return query;
   }
+
+  private isCostFilterValid(
+    fetchSpecification: ExtendedFetchSpecification,
+  ): boolean {
+    return !!(
+      fetchSpecification?.otherFilters?.costRange &&
+      fetchSpecification?.otherFilters?.costRange.length === 2 &&
+      fetchSpecification?.otherFilters?.costRange[0] <=
+        fetchSpecification?.otherFilters?.costRange[1] &&
+      fetchSpecification?.otherFilters?.costRangeSelector
```

Comment:
I am not sure as to why we need to validate costRangeSelector is present to check the costFilters. The cost filters should work independently to what cost selector is selected. Again, ideally we should be able to validate this with zod, let's check it out when we sync up.

But I don't really understand why we check both filters together